servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ClientParentConnectionContext.java

```diff
@@ -310,6 +352,11 @@ private void childChannelActive(Future<Http2StreamChannel> future,
                             unexpected.addSuppressed(cause);
                             LOGGER.warn("Unexpected exception while handling the original cause", unexpected);
                         }
+                    } else {
+                        streamObserver.streamClosed(cause);
```

Comment:
nit: consider putting the `streamObserver` and `onCloseRunnable` error recovery code in a shared method (unlikely to be called, reduce byte code of containing method).